javascript/javascript1/week3/main.js

```diff
@@ -0,0 +1,274 @@
+//Item array removal
+const names = [
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+const nameToRemove = "Ahmad";
+for (var i = 0; i < names.length; i++) {
+  if (names[i] === nameToRemove) {
+    names.splice(i, 1);
+    i--;
+  }
+}
+console.log(names);
+
+//When will we be there??
+const travelInformation = {
+  speed: 60,
+  destinationDistance: 567,
+};
+
+function travelTimeCalculation() {
+  var hours = (distance - (distance % speed)) / speed;
+  var minutes = ((distance % speed) * 60) / speed;
+  return [hours, minutes];
+}
+let distance = travelInformation.destinationDistance;
+let speed = travelInformation.speed;
+const travelTime = travelTimeCalculation(travelInformation);
```

Comment:
Your ultimate goal is to be able to put information into a function and get information out. As it is now, your function depends on `distance `and `speed`, both of which lie outside the function itself. You don't really want that. Everything necessary for the function to work should be inside it. 

I would suggest taking in `speed `and `distance `as parameters, because then you are not tied to the way the `travelInformation `object is structured. Who knows - maybe another user will have a `travelInformation `object that looks different and then you aren't sure if your function can use it or not. 

```suggestion
function travelTimeCalculation(distance ,speed ) {
  var hours = (distance - (distance % speed)) / speed;
  var minutes = ((distance % speed) * 60) / speed;
  return [hours, minutes];
}

const travelTime = travelTimeCalculation( 
  travelInformation.speed,
  travelInformation.destinationDistance);
```

Reply:
Yes, agree. Changed the parameter and made the function independent of distance and speed variable which are located outside the function.